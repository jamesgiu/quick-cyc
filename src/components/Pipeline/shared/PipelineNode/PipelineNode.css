.qc-node-wrapper {
    border-radius: 50%;
    display: inline-flex;
    padding: 20px;
    color: white;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    border: solid;
    position: relative;
    transition: background-color 0.5s, border 0.5s;
    z-index: 1;
}

.qc-node-percent-complete {
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    border-radius: 50%;
    z-index: 2;
    opacity: 0.3;
    transform: rotate(270deg);
}

.qc-node-wrapper.qc-intent-failure .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#cd4246 0 0) left/100% 0% no-repeat #ccc;
}

.qc-node-wrapper.qc-intent-none .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#5f6b7c 0 0) left/100% 0% no-repeat #ccc;
}

.qc-node-wrapper.qc-intent-in-progress .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#565b61 0 0) left/100% 0% no-repeat #ccc;
}

.qc-node-wrapper.qc-intent-success .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#238551 0 0) left/100% 0% no-repeat #ccc;
}

.qc-node-wrapper.qc-intent-warning .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#fbb360 0 0) left/100% 0% no-repeat #ccc;
}

.qc-node-wrapper.qc-intent-skipped .qc-node.qc-node-loading .qc-node-percent-complete {
    background: linear-gradient(#75787c 0 0) left/100% 0% no-repeat #ccc;
}

.qc-top-right-icon {
    border-radius: 50%;
    color: white;
    box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px, rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px, rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;
    position: absolute;
    text-align: center;
    vertical-align: middle;
    border: solid 0.1px;
    z-index: 3;
}

.qc-top-right-icon.qc-size-xs {
    width: 10px;
    height: 10px;
    padding: 10px;
    top: -10px;
    right: -10px;
}

.qc-top-right-icon.qc-size-s {
    width: 15px;
    height: 15px;
    padding: 10px;
    top: -10px;
    right: -10px;
}

.qc-top-right-icon.qc-size-m {
    width: 20px;
    height: 20px;
    padding: 10px;
    top: -10px;
    right: -10px;
    font-size: 1.2rem;
}

.qc-top-right-icon.qc-size-l {
    width: 30px;
    height: 30px;
    padding: 10px;
    top: -10px;
    right: -5px;
    font-size: 1.5rem;
}

.qc-top-right-icon.qc-size-xl {
    width: 35px;
    height: 35px;
    padding: 10px;
    top: -10px;
    right: -2px;
    font-size: 1.8rem;
}

.qc-node-icon {
    z-index: 2;
}

/* Sizes */
.qc-node-wrapper.qc-size-xs,
.qc-node-wrapper.qc-size-xs .qc-node-icon svg {
    width: 15px;
    height: 15px;
}

.qc-node-wrapper.qc-size-s,
.qc-node-wrapper.qc-size-s .qc-node-icon svg {
    width: 30px;
    height: 30px;
}

.qc-node-wrapper.qc-size-m,
.qc-node-wrapper.qc-size-m .qc-node-icon svg {
    width: 40px;
    height: 40px;
}

.qc-node-wrapper.qc-size-l,
.qc-node-wrapper.qc-size-l .qc-node-icon svg {
    width: 60px;
    height: 60px;
}

.qc-node-wrapper.qc-size-xl,
.qc-node-wrapper.qc-size-xl .qc-node-icon svg {
    width: 80px;
    height: 80px;
}

.qc-node-wrapper.qc-size-l,
.qc-node-wrapper.qc-size-xl {
    padding: 30px;
}