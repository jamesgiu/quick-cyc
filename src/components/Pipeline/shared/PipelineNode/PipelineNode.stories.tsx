import type { Meta, StoryObj } from "@storybook/react-webpack5";

import { PipelineIntent, PipelineSize } from "../../types";
import { PipelineIcons } from "../../icons";
import PipelineNode from "./PipelineNode";

const meta: Meta<typeof PipelineNode> = {
  title: "PipelineNode",
  component: PipelineNode,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
};

type Story = StoryObj<typeof PipelineNode>;

export const Primary: Story = {
  args: {
    onNodeClick: () => {
      console.log("Clicked!");
    },
    intent: PipelineIntent.IN_PROGRESS,
    active: true,
    icon: PipelineIcons.Icon2fa,
    size: PipelineSize.M,
    attempts: 2,
    percentComplete: 80,
    outerLabel: "Deploy",
  },
};

export default meta;
