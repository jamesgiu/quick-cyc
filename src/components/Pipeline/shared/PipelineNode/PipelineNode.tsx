import React from "react";
import { PipelineIntent, PipelineSize } from "../../types";
// @ts-ignore
import * as allIcons from "@tabler/icons-react";
import { PipelineIcons } from "../../icons";
import "./PipelineNode.css";
import "../../../index.css";

export interface PipelineNodeProps {
  intent: PipelineIntent;
  active?: boolean;
  icon: PipelineIcons;
  size: PipelineSize;
  percentComplete?: number;
  attempts?: number;
  onNodeClick?: () => void;
  innerLabel?: string;
  outerLabel?: string;
  className?: string;
}

export const PipelineNode = (props: PipelineNodeProps) => {
  const IconToBeUsed = allIcons[props.icon];

  return (
    <span
      className={
        "qc-node-wrapper" +
        ` ${props.className} ${props.size} ${props.intent} ${props.active ? "qc-state-active" : ""} ${props.onNodeClick !== undefined ? "qc-clickable" : ""}`
      }
    >
      <span
        className={
          "qc-node" +
          ` ${props.percentComplete && props.percentComplete < 100 ? "qc-node-loading" : ""}`
        }
      >
        {props.attempts !== undefined &&
          props.intent !== PipelineIntent.SKIPPED && (
            <span
              className={"qc-top-right-icon" + ` ${props.size} ${props.intent}`}
            >
              {props.attempts}
            </span>
          )}

        {props.percentComplete && (
          <span
            className={"qc-node-percent-complete"}
            style={{ backgroundSize: props.percentComplete + "%" }}
          />
        )}

        {props.intent === PipelineIntent.SKIPPED && (
          <span
            className={"qc-top-right-icon" + ` ${props.size} ${props.intent}`}
          >
            <allIcons.IconPlayerSkipForwardFilled size={18} />
          </span>
        )}
        <span className={"qc-node-icon"} onClick={props.onNodeClick}>
          <IconToBeUsed />
          {props.innerLabel && (
            <span className={"qc-node-label"}>{props.innerLabel}</span>
          )}
        </span>

        {props.outerLabel && (
          <div className={"qc-node-outer-label" + ` ${props.intent}`}>
            <span className="qc-node-outer-label-inner">
              {props.outerLabel}
            </span>
          </div>
        )}
      </span>
    </span>
  );
};

export default PipelineNode;
