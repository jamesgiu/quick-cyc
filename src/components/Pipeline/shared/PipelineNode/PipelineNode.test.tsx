// @vitest-environment jsdom
 
import { expect, test } from 'vitest';
 
// Replace your-framework with the framework you are using, e.g. react-vite, nextjs, nextjs-vite, etc.
import { composeStories } from '@storybook/react-vite';
 
import * as stories from './PipelineNode.stories';
 
const { Primary } = composeStories(stories);
test('Snapshot test', async () => {
  await Primary.run();
  expect(document.body.firstChild).toMatchSnapshot();
});