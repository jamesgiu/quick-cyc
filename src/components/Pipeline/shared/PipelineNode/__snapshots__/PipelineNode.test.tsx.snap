// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pipeline snapshot 1`] = `
<div>
  <span
    class="qc-node-wrapper undefined qc-size-m qc-intent-in-progress qc-state-active qc-clickable"
  >
    <span
      class="qc-node qc-node-loading"
    >
      <span
        class="qc-top-right-icon qc-size-m qc-intent-in-progress"
      >
        2
      </span>
      <span
        class="qc-node-percent-complete"
        style="background-size: 80%;"
      />
      <span
        class="qc-node-icon"
      >
        <svg
          class="tabler-icon tabler-icon-2fa"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 16h-4l3.47 -4.66a2 2 0 1 0 -3.47 -1.54"
          />
          <path
            d="M10 16v-8h4"
          />
          <path
            d="M10 12l3 0"
          />
          <path
            d="M17 16v-6a2 2 0 0 1 4 0v6"
          />
          <path
            d="M17 13l4 0"
          />
        </svg>
      </span>
      <div
        class="qc-node-outer-label qc-intent-in-progress"
      >
        <span
          class="qc-node-outer-label-inner"
        >
          Deploy
        </span>
      </div>
    </span>
  </span>
</div>
`;

exports[`Snapshot test 1`] = `
<div>
  <span
    class="qc-node-wrapper undefined qc-size-m qc-intent-in-progress qc-state-active qc-clickable"
  >
    <span
      class="qc-node qc-node-loading"
    >
      <span
        class="qc-top-right-icon qc-size-m qc-intent-in-progress"
      >
        2
      </span>
      <span
        class="qc-node-percent-complete"
        style="background-size: 80%;"
      />
      <span
        class="qc-node-icon"
      >
        <svg
          class="tabler-icon tabler-icon-2fa"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 16h-4l3.47 -4.66a2 2 0 1 0 -3.47 -1.54"
          />
          <path
            d="M10 16v-8h4"
          />
          <path
            d="M10 12l3 0"
          />
          <path
            d="M17 16v-6a2 2 0 0 1 4 0v6"
          />
          <path
            d="M17 13l4 0"
          />
        </svg>
      </span>
      <div
        class="qc-node-outer-label qc-intent-in-progress"
      >
        <span
          class="qc-node-outer-label-inner"
        >
          Deploy
        </span>
      </div>
    </span>
  </span>
</div>
`;
